streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -451,7 +451,7 @@ private[zio] final class ChannelExecutor[Env, InErr, InElem, InDone, OutErr, Out
     if (finalizers.isEmpty) null
     else
       ZIO
-        .foreach(finalizers)(_.apply(ex).exit)
+        .foreach(finalizers)(fin => provide(fin.apply(ex).exit))
```

Comment:
I think this will work too right? It'll be more performant because we're providing the environment just once in case there are multiple finalizers

```scala
      provide(ZIO.foreach(finalizers)(_.apply(ex).exit))
        .map(results => Exit.collectAll(results) getOrElse Exit.unit)
        .unexit
```

Reply:
I guess